pkgs/applications/science/logic/lean2/default.nix

```diff
@@ -1,19 +1,23 @@
-{ lib, stdenv, fetchFromGitHub, cmake, gmp, mpfr, python2
-, gperftools, ninja, makeWrapper }:
+{ lib, stdenv, fetchFromGitHub, cmake, gmp, mpfr, python3
+, jemalloc, ninja, makeWrapper }:
 
 stdenv.mkDerivation {
   pname = "lean2";
-  version = "2017-07-22";
+  version = "2018-10-01";
 
   src = fetchFromGitHub {
     owner  = "leanprover";
     repo   = "lean2";
-    rev    = "34dbd6c3ae612186b8f0f80d12fbf5ae7a059ec9";
-    sha256 = "1xv3j487zhh1zf2b4v19xzw63s2sgjhg8d62a0kxxyknfmdf3khl";
+    rev    = "8072fdf9a0b31abb9d43ab894d7a858639e20ed7";
+    sha256 = "12bscgihdgvaq5xi0hqf5r4w386zxm3nkx1n150lv5smhg8ga3gg";
   };
 
+  patches = [
+    # https://github.com/leanprover/lean2/pull/13
+    ./compilation-fix.patch
+  ];
```

Comment:
Can't we fetchpatch?

Reply:
Updated. Since my PR is still open, I might need to update the hash if upstream requests changes, but I can do that.